yarn-project/end-to-end/src/e2e_p2p_network.test.ts

```diff
@@ -154,11 +161,73 @@ describe('e2e_p2p_network', () => {
     );
 
     // shutdown all nodes.
-    for (const context of contexts) {
-      await context.node.stop();
-      await context.pxeService.stop();
+    await stopNodes(bootstrapNode, nodes);
+  });
+
+  // NOTE: If this test fails in a PR where the shuffling algorithm is changed, then it is failing as the node with
+  // the mocked p2p layer is being picked as the sequencer, and it does not have any transactions in it's mempool.
+  // If this is the case, then we should update the test to switch off the mempool of a different node.
+  // adjust `nodeToTurnOffTxGossip` in the test below.
+  it('should produce an attestation by requesting tx data over the p2p network', async () => {
+    /**
+     * Birds eye overview of the test
+     * 1. We spin up x nodes
+     * 2. We turn off receiving a tx via gossip from two of the nodes
+     * 3. We send a transactions and gossip it to other nodes
+     * 4. The disabled nodes will receive an attestation that it does not have the data for
+     * 5. They will request this data over the p2p layer
+     * 6. We receive all of the attestations that we need and we produce the block
+     *
+     * Note: we do not attempt to let this node produce a block, as it will not have received any transactions
```

Comment:
If we are running with walltime, this should not cause a failure but just a "long" wait because it will not publish, did you see so?

Reply:
I will experiment with this